src/scripts/main.js

```diff
@@ -1,3 +1,357 @@
+
+positionLabel.textContent = 'Position:';
+
+const positionInput = document.createElement('input');
+
+positionInput.setAttribute('type', 'text');
+positionInput.setAttribute('data-qa', 'position');
+positionInput.setAttribute('pattern', '[a-zA-Z]+');
+positionLabel.append(positionInput);
+form.appendChild(positionLabel);
+
+const selectLabel = document.createElement('label');
+
+selectLabel.textContent = 'Office:';
+
+const select = document.createElement('select');
+
+select.setAttribute('data-qa', 'office');
+select.setAttribute('required', '');
+selectLabel.append(select);
+form.appendChild(selectLabel);
+
+const offices = ['Tokyo', 'Singapore', 'London',
+  'New York', 'Edinburgh', 'San Francisco'];
+
+const selectForm = function() {
+  for (const town of offices) {
+    const option = document.createElement('option');
+
+    option.setAttribute('value', town);
+
+    const t = document.createTextNode(town);
+
+    option.appendChild(t);
+    select.appendChild(option);
+  }
+};
+
+selectForm();
+
+const ageLabel = document.createElement('label');
+
+ageLabel.textContent = 'Age:';
+
+const ageInput = document.createElement('input');
+
+ageInput.setAttribute('type', 'number');
+ageInput.setAttribute('data-qa', 'age');
+ageLabel.append(ageInput);
+form.appendChild(ageLabel);
+
+const salaryLabel = document.createElement('label');
+
+salaryLabel.textContent = 'Salary:';
+
+const salaryInput = document.createElement('input');
+
+salaryInput.setAttribute('type', 'number');
+salaryInput.setAttribute('data-qa', 'salary');
+
+function salaryCheck(value) {
+  if (salaryInput.value < 0) {
+    pushNotification('Error', 'Salary must be over 0.',
+      'error');
+  } else {
+    return true;
+  }
+}
+
+salaryLabel.append(salaryInput);
+form.appendChild(salaryLabel);
+
+const button = document.createElement('button');
+
+button.textContent = 'Save to table';
+form.appendChild(button);
+
+const arr = document.getElementsByTagName('input');
+
+[...arr].map(el => el.setAttribute('required', ''));
+
+const pushNotification = (title, description, type) => {
+  const notification = document.createElement('div');
+
+  notification.setAttribute('class', `notification ${type}`);
+  notification.setAttribute('data-qa', 'notification');
+
+  notification.insertAdjacentHTML('afterBegin',
+    `<h2 class = 'title'>${title}</h2><p>${description}</p>`);
+
+  document.body.append(notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 2000);
+};
+
+function validateForm(nameN, ageN) {
+  if (nameN.length < 4) {
+    pushNotification('Error', 'Name length must be more than 4 letters.',
+      'error');
+  } else if (ageN < 18 || ageN > 90) {
+    pushNotification('Error', 'Age must be over 18 but under 90.',
+      'error');
+  } else {
+    return true;
+  }
+};
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const nameV = document.getElementsByTagName('input')[0].value;
```

Comment:
```suggestion
  const name = document.getElementsByTagName('input')[0].value;
```

Reply:
const nameV is value of this cell, it is convenient for me to have such const, if it is not critical i would like to stay it (the same ageV, positionV...)

Reply:
Not resolved, you just need to rename this variable. Don't forget to check other cases.